File: Scala-Project/Task_2_DollarsConverter.scala
import scala.io.StdIn._

object Task_2_DollarsConverter {
  def main(args: Array[String]): Unit = {
    println("Enter the dollar amount :")
    val dollarAmount = readDouble()

    val rielAmount = dollarAmount * 4100

    val num2000 = (rielAmount / 2000).toInt
    val num1000 = ((rielAmount - (2000*num2000))/1000).toInt
    val num500 = ((rielAmount - (2000*num2000 + 1000*num1000))/500).toInt
    val num100 = ((rielAmount - (2000*num2000 + 1000*num1000 + 500*num500))/100).toInt

    println(s"There are $num2000: 2000 riel, $num1000: 1000 riel, $num500: 500 riel and $num100: 100 riel")
  }
}

File: Java-Project/Task_2_DollarsConverter.java
import java.util.*;
public class Task_2_DollarsConverter {
    public static void main(String[] args) {
        double dollarAmount, rielAmount;
        int num2000,num1000,num500,num100;
        Scanner aa = new Scanner(System.in);
        System.out.println("Enter the dollars amount :");
        dollarAmount = aa.nextDouble();
        rielAmount = dollarAmount * 4100;

        num2000 = (int)(rielAmount/2000);
        num1000 = (int) ((rielAmount -(2000*num2000))/1000);
        num500 = (int)((rielAmount -( 1000*num1000 + 2000*num2000))/500);
        num100 = (int)((rielAmount - (1000*num1000 + 2000*num2000 + 500*num500))/100);

        System.out.println(
                "There are "+num2000+" :2000r, "+num1000+" :1000r, "+num500+" :500r, "+num100+" :100r"
        );
    }
}